### PR TITLE
Adding Workspaces Reg and Deploy

### PR DESCRIPTION
**Pull request by Japeth**
- added to end of terraform deploy orb terraform output and persisted to a workspace. This will allow subsequent jobs to pick up on the output
- Pinned orb tools to specific version since the newest version is broken....
- Added Workspaces Orb
  - This installs the aws cli and jq tools
  - Orb checks for workspaces and workspaces_properties in the variables json
  - Orb will error out if any errors are encountered, EXCEPT if the error code is "ResourceExists.WorkSpaces"

### CODE REVIEW

File: src/jobs/terraform_deploy.yml [hunk 1]
```diff
@@ -52,14 +52,31 @@ steps:
...
+            terraform apply -input=false -auto-approve -var-file=<<parameters.var-file>> 
   - unless:
-      condition: <<parameters.var-file>>
+      condition: <<parameters.main-dir>>/<<parameters.var-file>>
       steps:
-        - run: terraform apply -input=false -auto-approve <<parameters.main-dir>>
- 
+        - run: |
+            cd <<parameters.main-dir>>
+            export TF_WORKSPACE=<<parameters.tf-workspace>>
```
Comment (murraybutler) [inline src/jobs/terraform_deploy.yml]: Can the exports be done as "environment" vars in the yaml?
https://circleci.com/docs/2.0/env-vars/#setting-an-environment-variable-in-a-step

**Review by murraybutler — DISMISSED** [review]
LGTM, just curious if the circleci environment keyword might be useful?

File: src/jobs/terraform_deploy.yml (hunk 1 above)
Comment (Japeth) [reply src/jobs/terraform_deploy.yml]: Sure.  I think that's conceivable

Comment (Japeth) [reply src/jobs/terraform_deploy.yml]: Code has been updated to use the environment keyword.  Tested and working.

Comment (murraybutler) [reply src/jobs/terraform_deploy.yml]: Sweet!!!  Nice work!!!

**Review by murraybutler — APPROVED** [review]
LGTM

**Pull request merged**